### PR TITLE
Reconnection delay of subscriber can be cancelled now

### PR DESCRIPTION
**Pull request by KonstantinRyazantsev**

### CODE REVIEW

File: src/Lykke.RabbitMqBroker/Subscriber/RabbitMqSubscriber.cs [hunk 1]
```diff
@@ -93,10 +94,10 @@ void IMessageConsumer<TTopicModel>.Subscribe(Func<TTopicModel, Task> callback)
 
         private bool IsStopped()
         {
-            return _thread == null;
+            return _cancellationTokenSource == null || _cancellationTokenSource.IsCancellationRequested;
         }
 
-        private void ReadThread()
+        private async void ReadThread()
```
Comment (starkmsu) [inline src/Lykke.RabbitMqBroker/Subscriber/RabbitMqSubscriber.cs]: Better put Task as returned type

File: src/Lykke.RabbitMqBroker/Subscriber/RabbitMqSubscriber.cs [hunk 2]
```diff
@@ -212,6 +213,18 @@ public RabbitMqSubscriber<TTopicModel> Start()
                 throw new InvalidOperationException("Please, specify log");
             }
 
+            if (_cancellationTokenSource == null)
```
Comment (starkmsu) [inline src/Lykke.RabbitMqBroker/Subscriber/RabbitMqSubscriber.cs]: may be it's better to merge conditions into one check - body of all execution paths is the same

File: src/Lykke.RabbitMqBroker/Subscriber/RabbitMqSubscriber.cs (hunk 1 above)
Comment (MikhailGerasimov) [reply src/Lykke.RabbitMqBroker/Subscriber/RabbitMqSubscriber.cs]: It's impossible. new Thread accept only this signature

Comment (KonstantinRyazantsev) [reply src/Lykke.RabbitMqBroker/Subscriber/RabbitMqSubscriber.cs]: As @MikhailGerasimov said, this is impossible. ReadThread - is entry point for the Thread.

File: src/Lykke.RabbitMqBroker/Subscriber/RabbitMqSubscriber.cs (hunk 2 above)
Comment (KonstantinRyazantsev) [reply src/Lykke.RabbitMqBroker/Subscriber/RabbitMqSubscriber.cs]: Agree

**Pull request merged**